python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -472,6 +472,8 @@ def mult(df):
             result.index.names = self.grouping.names
         else:
             result = cudf.concat(chunk_results)
+        if isinstance(self, SeriesGroupBy):
+            result.name = self.obj.name
```

Comment:
I would suggest otherwise... instead doing a runtime check here, we should probably overload `apply` in `SeriesGroupby`, take the result from super and apply name to it. Something like:
```python
def SeriesGroupby(Groupby):
    def apply(self, function):
        res = super().apply(function)
        res.name = self.obj.name
        return res
```